pkgs/development/compilers/fbc/default.nix

```
@@ -0,0 +1,165 @@
+    (fetchpatch {
+      name = "fbc-SOURCE_DATE_EPOCH-support.patch";
+      url = "https://github.com/freebasic/fbc/pull/392/commits/74ea6efdcfe9a90d1c860f64d11ab4a6cd607269.patch";
+      excludes = [ "changelog.txt" ];
+      sha256 = "sha256-v5FTi4vKOvSV03kigZDiOH8SEGEphhzkBL6p1hd+NtU=";
+    })
+  ];
+
+  postPatch = ''
+    patchShebangs tests/warnings/test.sh
+
+    # Some tests lack proper dependency on libstdc++
+    for missingStdcpp in tests/cpp/{class,call2}-fbc.bas; do
+      sed -i -e "/'"' TEST_MODE : /a #inclib "stdc++"' $missingStdcpp
+    done
+
+    # Help compiler find libstdc++ with gcc backend
+    sed -i -e '/fbcAddLibPathFor( "libgcc.a" )/a fbcAddLibPathFor( "libstdc++.so" )' src/compiler/fbc.bas
+  '';
+
+  dontConfigure = true;
+
+  depsBuildBuild = [
+    buildPackages.stdenv.cc
+    buildPackages.ncurses
+    buildPackages.libffi
+  ];
+
+  buildInputs = [
+    ncurses
+    libffi
+  ] ++ lib.optionals stdenv.hostPlatform.isLinux [
+    gpm
+    libGL
+    libX11
+    libXext
+    libXpm
+    libXrandr
+  ];
+
+  enableParallelBuilding = true;
+
+  hardeningDisable = [
+    "format"
+  ];
+
+  makeFlags = lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
+    "TARGET=${stdenv.hostPlatform.config}"
+  ];
+
+  preBuild = ''
+    export buildJobs=$NIX_BUILD_CORES
+    if [ -z "$enableParallelBuilding" ]; then
+      buildJobs=1
+    fi
+
+    echo Bootstrap an unpatched build compiler
+    make bootstrap-minimal -j$buildJobs \
+      BUILD_PREFIX=${buildPackages.stdenv.cc.targetPrefix} LD=${buildPackages.stdenv.cc.targetPrefix}ld
+
+    echo Compile patched build compiler and host rtlib
+    make compiler -j$buildJobs \
+      "FBC=$PWD/bin/fbc${stdenv.buildPlatform.extensions.executable} -i $PWD/inc" \
+      BUILD_PREFIX=${buildPackages.stdenv.cc.targetPrefix} LD=${buildPackages.stdenv.cc.targetPrefix}ld
+    make rtlib -j$buildJobs \
+      "FBC=$PWD/bin/fbc${stdenv.buildPlatform.extensions.executable} -i $PWD/inc" \
+      ${if (stdenv.buildPlatform == stdenv.hostPlatform) then
+        "BUILD_PREFIX=${buildPackages.stdenv.cc.targetPrefix} LD=${buildPackages.stdenv.cc.targetPrefix}ld"
+      else
+        "TARGET=${stdenv.hostPlatform.config}"
+      }
+
+    echo Install patched build compiler and host rtlib to local directory
+    make install-compiler prefix=$PWD/patched-fbc
+    make install-rtlib prefix=$PWD/patched-fbc ${lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) "TARGET=${stdenv.hostPlatform.config}"}
+    make clean
+
+    echo Compile patched host everything with previous patched stage
+    buildFlagsArray+=("FBC=$PWD/patched-fbc/bin/fbc${stdenv.buildPlatform.extensions.executable} -i $PWD/inc")
+  '';
+
+  installFlags = [
+    "prefix=${placeholder "out"}"
+  ];
+
+  doCheck = stdenv.buildPlatform == stdenv.hostPlatform;
+
+  checkTarget = lib.strings.concatStringsSep " " [
+    "unit-tests"
+    "warning-tests"
+    "log-tests"
+  ];
```

```suggestion
  checkTarget = "unit-tests warning-tests log-tests";
```
checkTarget is a string.